src/parsing/response.rs

```diff
@@ -156,26 +156,26 @@ impl Response {
         self.reader.text_with(charset)
     }
 
-    /// Create a `TextReader` from this `ResponseReader`.
+    /// Create a text reader from this `ResponseReader`.
     ///
     /// If the response headers contain charset information, that charset will be used to decode the body.
     /// Otherwise, if a default encoding is set it will be used. If there is no default encoding, ISO-8859-1
     /// will be used.
     ///
     /// This method only exists when the `charsets` feature is enabled.
     #[cfg(feature = "charsets")]
-    pub fn text_reader(self) -> TextReader<BufReader<ResponseReader>> {
+    pub fn text_reader(self) -> impl Read {
```

Comment:
I'm not a fan or returning `impl Read`. I'd like to return a `TextReader` that implements `Read` to we can add methods on it in the future if necessary.

Reply:
Will add back `TextReader` as a wrapper around `DecodeBytesReader`.

Reply:
Done.